src/util/tab-events.js

```diff
@@ -1,56 +1,55 @@
 import eventToPromise from './event-to-promise'
 
+const tabChangedEvents = [
+    {
+        event: browser.webNavigation.onCommitted,
+        filter: (details)=>(details.tabId == tabId && details.frameId == 0),
+        reason: "Tab URL changed before loading was complete"
+    },
+    {
+        event: browser.webNavigation.onHistoryStateUpdated,
+        filter: (details)=>(details.tabId == tabId && details.frameId == 0),
+        reason: "Tab URL changed before loading was complete"
+    },
+    {
+        event: browser.tabs.onRemoved,
+        filter: (closedTabId) => (closedTabId === tabId),
+        reason: {message: "Tab was closed before loading was complete."},
```

Comment:
These messages were different for each function, "before loading was complete" is a confusingly wrong message when printed by the whenTabActivated function. We can make them generic enough to match every of the functions, e.g. ".... before event occurred."
Also, it should be an object with a message field, not a string like you used above.